File: spec/test_user_code.c
#include "testparser.h"
#include <assert.h>
#include <stdio.h>
#include <string.h>

int main()
{
    char const * input = "abcdef";
    p_context_t context;
    p_context_init(&context, (uint8_t const *)input, strlen(input));
    assert(p_parse(&context) == P_SUCCESS);
    printf("pass1\n");

    input = "abcabcdef";
    p_context_init(&context, (uint8_t const *)input, strlen(input));
    assert(p_parse(&context) == P_SUCCESS);
    printf("pass2\n");

    return 0;
}

File: spec/test_user_code.rs
use testparser::*;

fn main() {
    let mut context = p_context_new(b"abcdef");
    assert_eq!(P_SUCCESS, p_parse(&mut context));
    println!("pass1");
    p_context_delete(context);

    let mut context = p_context_new(b"abcabcdef");
    assert_eq!(P_SUCCESS, p_parse(&mut context));
    println!("pass2");
    p_context_delete(context);
}
